api/src/jmh/java/io/opentelemetry/api/trace/BaggageBenchmarks.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.trace;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageBuilder;
+import io.opentelemetry.context.Context;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+@SuppressWarnings("JavadocMethod")
+@State(Scope.Thread)
+public class BaggageBenchmarks {
+
+  @Param({"0", "1", "10", "100"})
+  public int itemsToAdd;
+
+  @Benchmark
+  @BenchmarkMode({Mode.AverageTime})
+  @Fork(1)
+  @Measurement(iterations = 15, time = 1)
+  @OutputTimeUnit(TimeUnit.NANOSECONDS)
+  @Warmup(iterations = 5, time = 1)
+  public Baggage baggageItemBenchmark() {
+    BaggageBuilder builder = Baggage.builder();
+    for (int i = 0; i < itemsToAdd; i++) {
+      builder.put("key" + i, "value" + i);
```

Comment:
Can you preallocate the key / value strings in a `Setup`? No idea how significant the string allocation here is without trying that, it may not be though. Though it would definitely skew the GC numbers

Reply:
sure thing. will do so!

Reply:
updated...it didn't materially change the benchmarks much.